monitoring/grafana/dashboards/tests/util.py

```diff
@@ -0,0 +1,43 @@
+import json
+import os
+from typing import Any, Dict, Optional, Tuple, Union
+
+UNITS = ['ms', 's', 'm', 'h', 'd', 'w', 'y']
+
+def resolve_time_and_unit(time: str) -> Union[Tuple[int, str], Tuple[None, None]]:
+    """
+    Divide time with its unit and return a tuple like (10, 'm')
+    Return None if its and invalid prometheus time
+    Valid units are inside UNITS.
+    """
+    if time[-1] in UNITS:
+        return int(time[:-1]), time[-1]
+    if time[-2:] in UNITS:
+        return int(time[:-2]), time[-2:]
+    return None, None
+
+def construct_query_map() -> Dict[str, Any]:
+    pwd = os.getcwd()
+    files = os.listdir(pwd)
+    query_map = {}
+    for file in files:
+        if file.endswith('.json'):
+            with open(os.path.join(pwd, file), 'r') as f:
+                dashboard_data = json.loads(f.read())
+                if 'panels' not in dashboard_data:
+                    continue
+                for panel in dashboard_data['panels']:
+                    if (
+                            'title' in panel
+                            and 'targets' in panel
+                            and len(panel['targets']) > 0
+                            and 'expr' in panel['targets'][0]
+                    ):
+                        query_map[panel['title']] = panel['targets'][0]['expr']
```

Comment:
BTW, while testing that I realized that some panels may have more than 1 expression, like `pools-details.json`>`$pool_name Client IOPS` (read and write):
* `irate(ceph_pool_rd[1m]) * on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}`
* `irate(ceph_pool_wr[1m]) * on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}`

Not sure if you are familiar with `jq` CLI (or other 'pseudo-languages' for inspecting structured data like XPath/JSONPath/etc), but this can also be done like this:
```py
import jq

query_map = jq.compile('[.panels[] | {(.title): .targets}] | add').input(text=file.read()).all()
```
```json
{
  "Capacity used": [
    {
      "expr": "(ceph_pool_stored / (ceph_pool_stored + ceph_pool_max_avail)) * on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}",
      "format": "time_series",
      "intervalFactor": 1,
      "legendFormat": "",
      "refId": "A"
    }
  ],
  "Time till full": [
    {
      "expr": "(ceph_pool_max_avail / deriv(ceph_pool_stored[6h])) * on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"} > 0",
      "format": "time_series",
      "intervalFactor": 1,
      "legendFormat": "",
      "refId": "A"
    }
  ],
  "$pool_name Object Ingress/Egress": [
    {
      "expr": "deriv(ceph_pool_objects[1m]) * on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}",
      "format": "time_series",
      "intervalFactor": 1,
      "legendFormat": "Objects per second",
      "refId": "A"
    }
  ],
  "$pool_name Client IOPS": [
    {
      "expr": "irate(ceph_pool_rd[1m]) * on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}",
      "format": "time_series",
      "intervalFactor": 1,
      "legendFormat": "reads",
      "refId": "A"
    },
    {
      "expr": "irate(ceph_pool_wr[1m]) * on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}",
      "format": "time_series",
      "intervalFactor": 1,
      "legendFormat": "writes",
      "refId": "B"
    }
  ],
  "$pool_name Client Throughput": [
    {
      "expr": "irate(ceph_pool_rd_bytes[1m]) + on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}",
      "format": "time_series",
      "intervalFactor": 1,
      "legendFormat": "reads",
      "refId": "A"
    },
    {
      "expr": "irate(ceph_pool_wr_bytes[1m]) + on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}",
      "format": "time_series",
      "intervalFactor": 1,
      "legendFormat": "writes",
      "refId": "B"
    }
  ],
  "$pool_name Objects": [
    {
      "expr": "ceph_pool_objects * on(pool_id) group_left(instance,name) ceph_pool_metadata{name=~\"$pool_name\"}",
      "format": "time_series",
      "intervalFactor": 1,
      "legendFormat": "Number of Objects",
      "refId": "A"
    }
  ]
}
```

So the above map would allow to write statements like:
```behave
Then Grafana panel "$pool_name Client IOPS" with legend "reads":
```
Or something more compact:
```behave
Then Grafana panel "$pool_name Client IOPS (reads)":
```